sql/sys_vars.cc

```diff
@@ -4414,7 +4414,43 @@ void init_log_slow_verbosity()
 {
   update_slow_query_log_use_global_control(0,0,OPT_GLOBAL);
 }
-static Sys_var_set Sys_slow_query_log_use_global_control(
+
+/**
+  Specialized class that handles "none" value of
+  slow_query_log_use_global_control_set variable.
+  When "none" only value is detected, it is rewriten to empty
+  causing set to be cleared.
+*/
+class Sys_var_set_none: public Sys_var_set {
```

Comment:
Shouldn't this go into the header, where other classes like this are placed?

Reply:
I think no. This specialization is only for this particular variable and there is no need to make this type containing fix public.
Of course we can imagine more sophisticated general purpose class like this, that allows us to provide list of possible values that should behave like 'none' like vector<string> = {'none', 'null', 'nothing'}. 
But in this case I would keep it here because there is no need to make it public.